controllers/userController.js

```diff
@@ -25,18 +34,49 @@ const loginUser = async (req, res) => {
 const registerUser = async (req, res) => {
   const { email, password } = req.body;
 
+  // create a token
+  const token = createToken(email);
+
   try {
-    const user = await User.register(email, password);
+    const user = await User.register(email, password, token);
 
-    // create a token
-    const token = createToken(user._id);
+    const userName = user.email.split('@')[0];
 
-    res.status(200).json({ email, id: user._id, token, theme: user.theme, createdAt: user.createdAt });
+    transport.sendMail({
+      from: process.env.USER,
+      to: user.email,
+      subject: 'Please confirm your account',
+      html: `<div>
+               <h1>Email Confirmation</h1>
+               <h2>Hello ${userName}</h2>
+               <p>Thank you for subscribing. Please confirm your email by clicking on the following link</p>
+               <a href=http://localhost:3000/confirm/${user.confirmationCode}>Click here</a>
```

Comment:
wouldn't be localhost in production, so maybe save the url this should run at in an environmental variable that changes in production and development environments